nexus/src/authn/external/spoof.rs

```diff
@@ -13,21 +13,59 @@ use crate::authn::Actor;
-/// Magic header value to produce a "no such actor" error
-pub const SPOOF_RESERVED_BAD_ACTOR: &str = "Jack Donaghy";
-/// Magic header value to produce a "bad credentials" error
-pub const SPOOF_RESERVED_BAD_CREDS: &str =
-    "this fake I.D., it is truly excellent";
+// This scheme is intended for demos, development, and testing until we have a
+// more automatable identity provider that can be used for those purposes.
+//
+// For ease of integration into existing clients, we use RFC 6750 bearer tokens.
+// This mechanism in turn uses HTTP's "Authorization" header.  In practice, it
+// looks like this:
+//
+//     Authorization: Bearer oxide-spoof-001de000-05e4-4000-8000-000000060001
+//     ^^^^^^^^^^^^^  ^^^^^^ ^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+//     |              |      |           |
+//     |              |      |           +--- specifies the actor id
+//     |              |      +--------------- specifies this "spoof" mechanism
+//     |              +---------------------- specifies RFC 6750 bearer tokens
+//     +------------------------------------- standard HTTP authentication hdr
+//
+// (That's not a typo -- the "authorization" header is generally used to specify
+// _authentication_ information.  Similarly, the "Unauthorized" HTTP response
+// code usually describes an _authentication_ error.)
+//
+// **This mechanism trusts (without verification) that the client is whoever
+// they say they are.**
+
 pub const SPOOF_SCHEME_NAME: authn::SchemeName = authn::SchemeName("spoof");
+/// Header used for "spoof" authentication
+pub const HTTP_HEADER_OXIDE_AUTHN_SPOOF: &http::header::HeaderName =
+    &http::header::AUTHORIZATION;
+
+/// Magic value to produce a "no such actor" error
+const SPOOF_RESERVED_BAD_ACTOR: &str = "Jack-Donaghy";
+/// Magic value to produce a "bad credentials" error
+const SPOOF_RESERVED_BAD_CREDS: &str = "this-fake-ID-it-is-truly-excellent";
+/// Prefix used on the bearer token to identify this scheme
+// RFC 6750 expects bearer tokens to be opaque base64-encoded data.  In our
+// case, the data we want to represent (this prefix, plus valid uuids) are
+// subsets of the base64 character set, so we do not bother encoding them.
+const SPOOF_PREFIX: &str = "oxide-spoof-";
 
 lazy_static! {
+    /// Actor (id) used for the special "bad credentials" error
     static ref SPOOF_RESERVED_BAD_CREDS_ACTOR: Actor =
         Actor("22222222-2222-2222-2222-222222222222".parse().unwrap());
+    /// Complete HTTP header value to trigger the "bad actor" error
+    pub static ref SPOOF_HEADER_BAD_ACTOR: http::header::HeaderValue =
+        make_header_value_raw(SPOOF_RESERVED_BAD_ACTOR.as_bytes()).unwrap();
+    /// Complete HTTP header value to trigger the "bad creds" error
+    pub static ref SPOOF_HEADER_BAD_CREDS: http::header::HeaderValue =
+        make_header_value_raw(SPOOF_RESERVED_BAD_CREDS.as_bytes()).unwrap();
 }
 
 /// Implements a (test-only) authentication scheme where the client simply
```

Comment:
`custom header` in the line below this one is no longer accurate

Reply:
Good point.  Fixed.